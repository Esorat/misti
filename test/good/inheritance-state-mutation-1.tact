trait T {
  balance: Int;
}

contract C with T {
  balance: Int = 42;
  fun updateBalance() {
    self.balance = 100; // Suspicious: Highlighted by the detector
  }
}
