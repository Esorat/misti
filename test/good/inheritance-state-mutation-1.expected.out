test/good/inheritance-state-mutation-1.tact:25:5:
  24 |     // Bad cases
> 25 |     self.balance = 100;
           ^
  26 |     self.balance -= 100;
Inherited trait field balance is mutated: Directly modifying inherited trait fields can indicate a potential error or poor design
Help: Consider using setter methods to preserve invariants
See: https://nowarp.io/tools/misti/docs/detectors/InheritedStateMutation

test/good/inheritance-state-mutation-1.tact:26:5:
  25 |     self.balance = 100;
> 26 |     self.balance -= 100;
           ^
  27 |     self.m.set(3, 3);
Inherited trait field balance is mutated: Directly modifying inherited trait fields can indicate a potential error or poor design
Help: Consider using setter methods to preserve invariants
See: https://nowarp.io/tools/misti/docs/detectors/InheritedStateMutation

test/good/inheritance-state-mutation-1.tact:27:5:
  26 |     self.balance -= 100;
> 27 |     self.m.set(3, 3);
           ^
  28 |     self.m.del(3);
Inherited trait field m is mutated: Directly modifying inherited trait fields can indicate a potential error or poor design
Help: Consider using setter methods to preserve invariants
See: https://nowarp.io/tools/misti/docs/detectors/InheritedStateMutation

test/good/inheritance-state-mutation-1.tact:28:5:
  27 |     self.m.set(3, 3);
> 28 |     self.m.del(3);
           ^
  29 |     self.a.x = 42;
Inherited trait field m is mutated: Directly modifying inherited trait fields can indicate a potential error or poor design
Help: Consider using setter methods to preserve invariants
See: https://nowarp.io/tools/misti/docs/detectors/InheritedStateMutation

test/good/inheritance-state-mutation-1.tact:29:5:
  28 |     self.m.del(3);
> 29 |     self.a.x = 42;
           ^
  30 |     self.a.x += 42;
Inherited trait field a is mutated: Directly modifying inherited trait fields can indicate a potential error or poor design
Help: Consider using setter methods to preserve invariants
See: https://nowarp.io/tools/misti/docs/detectors/InheritedStateMutation

test/good/inheritance-state-mutation-1.tact:30:5:
  29 |     self.a.x = 42;
> 30 |     self.a.x += 42;
           ^
  31 |     self.a.m.set(3, 3);
Inherited trait field a is mutated: Directly modifying inherited trait fields can indicate a potential error or poor design
Help: Consider using setter methods to preserve invariants
See: https://nowarp.io/tools/misti/docs/detectors/InheritedStateMutation

test/good/inheritance-state-mutation-1.tact:31:5:
  30 |     self.a.x += 42;
> 31 |     self.a.m.set(3, 3);
           ^
  32 |     self.a.m.del(3);
Inherited trait field a is mutated: Directly modifying inherited trait fields can indicate a potential error or poor design
Help: Consider using setter methods to preserve invariants
See: https://nowarp.io/tools/misti/docs/detectors/InheritedStateMutation

test/good/inheritance-state-mutation-1.tact:32:5:
  31 |     self.a.m.set(3, 3);
> 32 |     self.a.m.del(3);
           ^
  33 |   }
Inherited trait field a is mutated: Directly modifying inherited trait fields can indicate a potential error or poor design
Help: Consider using setter methods to preserve invariants
See: https://nowarp.io/tools/misti/docs/detectors/InheritedStateMutation